app/controllers/works_controller.rb

```diff
@@ -0,0 +1,84 @@
+class WorksController < ApplicationController
+  def index
+    @works = Work.all
+    @movies = Work.sort_by_category("movie")
+    @books = Work.sort_by_category("book")
+    @albums = Work.sort_by_category("album")
+  end
+
+  def show
+    work_id = params[:id]
+    @work = Work.find_by(id: work_id)
+
+    if @work.nil?
+      flash[:warning] = "Invalid Media."
+      redirect_to works_path
+      return
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+    if @work.save
+      flash[:success] = "Media added successfully"
+      redirect_to work_path(@work.id)
+      return
+    else
+      flash[:warning] = "Oops! Something is wrong. Cannot add Media."
+      redirect_to works_path
+      return
+    end
+  end
+
+  def edit
+    @work = Work.find_by(id: params[:id])
+
+    if @work.nil?
+      flash[:warning] = "Invalid request. Media not found."
+      redirect_to works_path
+      return
+    end
+  end
+
+  def update
+    @work = Work.find_by(params[id])
```

Comment:
`find_by` requires key-value pairs

```suggestion
    @work = Work.find_by(id: params[id])
```